File: MyBooks/MyBooks.Model/Znacka.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyBooks.Model
{
    public class Znacka
    {
        public int Id { get; set; }
        public string Naziv { get; set; } = null!;
        public string? Opis { get; set; }
        public string? Ikonica { get; set; }
        public string Tip { get; set; } = null!;
        public int Prag { get; set; }
        public int? Nivo { get; set; }
        public int TrenutniNapredak { get; set; }   // novo

        // NOVO
        public int Preostalo => Math.Max(0, Prag - TrenutniNapredak);

        public double Procenat =>
            Prag == 0
                ? 0
                : Math.Min(1.0, (double)TrenutniNapredak / Prag);
    }
}

File: mybooks_mobile/lib/models/znacka.dart
import 'package:json_annotation/json_annotation.dart';

part 'znacka.g.dart';

@JsonSerializable()
class Znacka {
  int id;
  String naziv;
  String? opis;
  String? ikonica;
  String tip;
  int prag;
  int? nivo;
  int? trenutniNapredak;
  Znacka(this.id, this.naziv, this.opis, this.ikonica, this.tip, this.prag,this.nivo,this.trenutniNapredak);

  factory Znacka.fromJson(Map<String, dynamic> json) => _$ZnackaFromJson(json);

  Map<String, dynamic> toJson() => _$ZnackaToJson(this);
}
